drafts/2.4-representable-functors.md

```diff
@@ -2,30 +2,34 @@
 type Reader a x = a -> x
 ```
 ```kotlin:ank:silent
-
+typealias Reader<A, X> = (A) -> X
 ```
 ................
 ```Haskell
 instance Functor (Reader a) where
     fmap f h = f . h
 ```
 ```kotlin:ank:silent
-
+fun <D, A> fmap(run: (D) -> A): Reader<D, A> = Kleisli(run.andThen { Id(it) })
 ```
 ................
 ```Haskell
 type Op a x = x -> a
 ```
 ```kotlin:ank:silent
-
+typealias OP<A,X> = (X)-> A
 ```
 ................
 ```Haskell
 instance Contravariant (Op a) where
     contramap f h = h . f
 ```
-```kotlin:ank:silent
-
+```kotlin:ank:playground
+import arrow.core.compose
+ 
+data class Op<A, B> constructor(private val f: (A) -> B) {
```

Comment:
I saw in the section [natural transformations](https://github.com/arrow-kt/Category-Theory-for-Programmers.kt/blob/master/src/main/ank/1.10-natural-transformations.md) :
```haskell
instance Contravariant (Op r) where
  contramap f (Op g) = Op (g . f)
```
```kotlin
import arrow.core.compose

data class Op<A, B> constructor(private val f: (A) -> B) {
  fun <C> contramap(op: Op<B, C>) = Op<A, C> { (op.f compose f)(it) }
}
```
are they different?

Reply:
Yes drastically. 
Refer to your implementation a few lines below. 
PIng me in Slack if you need help

Reply:
@arioston  Thank you for pointing out the error [here](https://github.com/arrow-kt/Category-Theory-for-Programmers.kt/blob/master/src/main/ank/1.10-natural-transformations.md). I will correct it in the next PR

Reply:
@i-walker is this the correct implementation ?
```kotlin
interface OpContravariant<C>: Contravariant<Conested<ForOp, C>> {
    override fun <A, B> Kind<Conested<ForOp, C>, A>.contramap(f: (B) -> A): Kind<Conested<ForOp, C>, B> =
         counnest().fix().compose(f.k()).conest()
}
```

Reply:
This may look like this:
```
@higherkind
data class Op<A, X>(val f: (X) -> A) {
  fun <B> compose(other: Op<B, A>): Op<B, X> =
    Op(other.f compose f)

  fun <B> compose(g: (A) -> B): Op<B, X> =
    Op(g compose f)
}
```
I wrote 2 isomorphic representations of `compose`. You can choose, which one makes more sense. I prefer the one without the wrapper

Reply:
Thanks to @1Jajen1 the contravariant instance may look like this:
```
interface OpContravariant<I>: Contravariant<OpPartialOf<I>> {
  override fun <A, B> Kind<OpPartialOf<I>, A>.contramap(f: (B) -> A): Kind<OpPartialOf<I>, B> =
    Op(fix().f compose f)
}
```